src/corelibs/U2Gui/src/Notification.cpp

```diff
@@ -281,32 +211,22 @@ QAction* Notification::getAction() const {
 void Notification::incrementCounter() {
     counter++;
     timestamp = QDateTime::currentMSecsSinceEpoch();
-
     updateDisplayText();
-
-    // Re-set on-screen time timeout.
-    onScreenStartTimeMillis = QDateTime::currentMSecsSinceEpoch();
-}
-
-qint64 Notification::getOnScreenStartTimeMillis() const {
-    return onScreenStartTimeMillis;
-}
-
-bool Notification::isOnScreen() const {
-    return onScreenTimer.isActive();
 }
 
 NotificationStack::NotificationStack(QWidget* _parentWidget)
     : QObject(_parentWidget), parentWidget(_parentWidget) {
     SAFE_POINT(parentWidget, "Parent widget is null!", );
     notificationWidget = new NotificationWidget(parentWidget);
     parentWidget->installEventFilter(this);
+
+    onScreenTimer.setInterval(100);
+    connect(&onScreenTimer, &QTimer::timeout, this, &NotificationStack::sl_updateNotificationState);
+    onScreenTimer.start();
 }
 
 NotificationStack::~NotificationStack() {
-    foreach (Notification* notification, notifications) {
-        delete notification;
-    }
+    qDeleteAll(floatingNotifications);
```

Comment:
You stopped deleting usual `notifications`

Reply:
Yes, this is correct behaviour